File: core/src/test/java/org/biacode/escommons/core/component/impl/ClockComponentImplTest.java
package org.biacode.escommons.core.component.impl;

import org.biacode.escommons.core.component.ClockComponent;
import org.biacode.escommons.core.test.AbstractCoreUnitTest;
import org.easymock.TestSubject;
import org.joda.time.DateTime;
import org.junit.Test;

import static org.junit.Assert.assertNotNull;

/**
 * Created by Arthur Asatryan.
 * Date: 7/14/17
 * Time: 5:07 PM
 */
public class ClockComponentImplTest extends AbstractCoreUnitTest {

    //region Test subject and mocks
    @TestSubject
    private ClockComponent clockComponent = new ClockComponentImpl();
    //endregion

    //region Constructors
    public ClockComponentImplTest() {
    }
    //endregion

    //region Test methods
    @Test
    public void testPrintCurrentDateTimeWithPredefinedFormatter() {
        // Test data
        // Reset
        resetAll();
        // Expectations
        // Replay
        replayAll();
        // Run test scenario
        final String currentDateTime = clockComponent.printCurrentDateTimeWithPredefinedFormatter();
        assertNotNull(currentDateTime);
        // Verify
        verifyAll();
    }

    @Test
    public void testGetCurrentDateTime() {
        // Test data
        // Reset
        resetAll();
        // Expectations
        // Replay
        replayAll();
        // Run test scenario
        final DateTime currentDateTime = clockComponent.getCurrentDateTime();
        assertNotNull(currentDateTime);
        // Verify
        verifyAll();
    }
    //endregion
}

File: escommons-toolkit/src/test/kotlin/org/biacode/escommons/toolkit/component/impl/ClockComponentImplTest.kt
package org.biacode.escommons.toolkit.component.impl

import org.assertj.core.api.Assertions.assertThat
import org.biacode.escommons.toolkit.test.AbstractEsCommonsToolkitUnitTest
import org.easymock.TestSubject
import org.junit.Test

/**
 * Created by Arthur Asatryan.
 * Date: 3/27/18
 * Time: 5:30 PM
 */
class ClockComponentImplTest : AbstractEsCommonsToolkitUnitTest() {

    //region Test subject and mocks
    @TestSubject
    private val clockComponent = ClockComponentImpl()
    //endregion

    //region Test methods
    @Test
    fun `test printCurrentDateTimeWithPredefinedFormatter`() {
        // test data
        resetAll()
        // expectations
        replayAll()
        // test scenario
        assertThat(clockComponent.printCurrentDateTimeWithPredefinedFormatter()).isNotNull().isNotBlank()
        verifyAll()
    }
    //endregion

}